clojure-mode.el

```diff
@@ -352,6 +352,21 @@ Called by `imenu--generic-function'."
               (set-match-data (list def-beg def-end)))))
         (goto-char start)))))
 
+(eval-and-compile
+  (defconst clojure-sym-rest-chars "^][\";\'@\\^`~\(\)\{\}\\"
+    "A black list of chars a clojure symbol must not contain. See
+definiton of 'macros' https://github.com/clojure/clojure/blob/master/src/jvm/clojure/lang/LispReader.java#L88")
+  (defconst clojure-sym-1st-chars (concat clojure-sym-rest-chars "0-9")
+    "A black list of chars a clojure symbol must not start with. See
+getMacro invocation:
+https://github.com/clojure/clojure/blob/master/src/jvm/clojure/lang/LispReader.java#L260
```

Comment:
This links to a function invocation, at least to me this line:

 `IFn macroFn = getMacro(ch);`

doesn't shed much light on anything discussed in this PR.